### PR TITLE
#150400956: implement voteCount in contribution comments

### PR DESCRIPTION
**Pull request by georgreen**
#### What does this PR do?
- Add voteCount field to the response body of: contributions/:id/comments
- Implement test for this feature
- Add pep8 config file: setup.cfg
- Fix pep8 errors in models, helpers, main , contributions and test_contributions

#### Description of Task to be completed?
- As a dev I should know how many upvotes a comment has for a contribution

#### How should this be manually tested?
run: ```$ pytest``` in projects root folder
run: ```$ pep8  path/<name of file> ``` in project root folder

#### What are the relevant pivotal tracker stories?
- #150400956

### CODE REVIEW

File: api/helpers.py [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
-    thread_comment_serialized = comment.serialize()
-    thread_comment_serialized["replies"] = [reply.serialize() for reply in comment.replies]
+    thread_comment_serialized = serialize_comment(comment)
+    thread_comment_serialized["replies"] = [serialize_comment(reply)
+                                            for reply in comment.replies]
     return thread_comment_serialized
 
-def save_tags(data, contribution):
 
+def serialize_comment(comment):
```
Comment (andela-oowonikoko) [inline api/helpers.py]: I don't think this method is necessary, the `serialize_thread_comments` can accommodate this changes.

File: api/helpers.py [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
+
 def validate_type(item, input_type):
     return type(item) is input_type
 
+
 def serialize_thread_comments(comment):
-    thread_comment_serialized = comment.serialize()
-    thread_comment_serialized["replies"] = [reply.serialize() for reply in comment.replies]
+    thread_comment_serialized = serialize_comment(comment)
+    thread_comment_serialized["replies"] = [serialize_comment(reply)
```
Comment (andela-oowonikoko) [inline api/helpers.py]: Have you considered having this method like this?

```
def serialize_thread_comments(comment):
    thread_comment_serialized = comment.serialize()
    thread_comment_serialized["voteCount"] = comment.votes.count()
    thread_comment_serialized["replies"] = [dict(reply.serialize(), 
        voteCount = reply.votes.count()) for reply in comment.replies]
    return thread_comment_serialized
```

File: main.py [hunk 3]
```diff
@@ -26,47 +26,62 @@
...
     @app.route('/')
     def index():
         return "Welcome to the Learning map Api"
 
-    # create endpoints
-    api = Api(app)
     api.add_resource(ContributionsResource,
-                     '/api/v1/contributions', '/api/v1/contributions',
+                     '/api/v1/contributions',
+                     '/api/v1/contributions',
```
Comment (andela-oowonikoko) [inline main.py]: Thank you for fixing all the pep8 errors!!! 
Can you please add a `/` in front of the `/api/v1/contributions` on line 55

File: test/api/views/test_contributions.py [hunk 4]
```diff
@@ -1179,21 +1204,63 @@ def test_update_comments_with_empty_content(self):
...
 
+    def test_comment_vote_count(self):
+
+        contribution = Contribution(
+            id="contrib_id",
+            title="test title",
+            description="test description",
+            contribution_type_id=self.contribution_type_id,
+            user_id="test")
+        db.session.add(contribution)
```
Comment (andela-oowonikoko) [inline test/api/views/test_contributions.py]: Please change all the `db.session.add() db.session.commit()` to `item.save()` as we have a `save` method in the models.

Comment (georgreen) [inline test/api/views/test_contributions.py]: Okay that can be done.

File: api/helpers.py (hunk 2 above)
Comment (georgreen) [inline api/helpers.py]: Yes I did but I was of the opinion that a function should only have one purpose, the above approach gives this function more than one purpose: serialize comments and make threaded comments, factoring out serializing comments seems to be a cleaner approach to this.

Comment (andela-oowonikoko) [inline api/helpers.py]: I believe we can rename the function as `serailize_comments`, so the function can serialize comments and their threads. So we are sure it's just a function that serializes comments.

**Comment by andela-oowonikoko** [comment]
LGTM

**Pull request merged**